RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByIsolation.cc

```diff
@@ -195,10 +228,29 @@ class PFRecoTauDiscriminationByIsolation : public PFTauDiscriminationProducerBas
   double offsetRelativeSumPt_;
   double customIsoCone_;
   
+  bool applyPhotonPtSumOutsideSignalConeCut_;
+  double maxAbsPhotonSumPt_outsideSignalCone_;
+  double maxRelPhotonSumPt_outsideSignalCone_;
+
+  bool applyFootprintCorrection_;
+  struct footprintCorrectionType
+  {
+    footprintCorrectionType(const std::string& selection, const std::string& offset)
+      : selection_(selection),
+	offset_(offset)
+    {}
+    ~footprintCorrectionType() {}
+    StringCutObjectSelector<PFTau> selection_;
+    StringObjectFunction<PFTau> offset_;
+  };
+  std::vector<footprintCorrectionType*> footprintCorrections_;
```

Comment:
How are smart pointers handled in member initializer lists? I'm now seeing compiler errors in this block of code:

RecoTauPiZeroStripPlugin3::RecoTauPiZeroStripPlugin3(const edm::ParameterSet& pset, edm::ConsumesCollector &&iC)
  : RecoTauPiZeroBuilderPlugin(pset, std::move(iC)),
    vertexAssociator_(pset.getParameter<edm::ParameterSet>("qualityCuts"), std::move(iC)),
    qcuts_(0),
    etaAssociationDistance_(0),
    phiAssociationDistance_(0)
(the last line, 118, specifically)

qcuts_, etaAssociationDistance_, phiAssociationDistance_ are all smart pointers

It throws this compiler error:
/afs/cern.ch/work/a/ajohnson/CMSSW_7_6_X_2015-08-04-2300/src/RecoTauTag/RecoTau/plugins/RecoTauPiZeroStripPlugin3.cc:118:30: error: call of overloaded 'unique_ptr(int)' is ambiguous
     phiAssociationDistance_(0)